### PR TITLE
Editorial: Fix a couple of inconsistencies

### PR DESCRIPTION
**Pull request by linusg**
Found while implementing these, details in commit messages.

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -33772,7 +33772,7 @@ <h1>Date.prototype.toISOString ( )</h1>
           1. Let _dateObject_ be the *this* value.
           1. Perform ? RequireInternalSlot(_dateObject_, [[DateValue]]).
           1. Let _tv_ be _dateObject_.[[DateValue]].
-          1. If _tv_ is not finite, throw a *RangeError* exception.
+          1. If _tv_ is *NaN*, throw a *RangeError* exception.
```
Comment (michaelficarra) [inline spec.html:33775]: If we were to make this change, I would probably add

```suggestion
          1. If _tv_ is *NaN*, throw a *RangeError* exception.
          1. Assert: _tv_ is an integral Number.
```

File: spec.html [hunk 2]
```diff
@@ -39569,18 +39571,16 @@ <h1>Array.prototype.toLocaleString ( [ _reserved1_ [ , _reserved2_ ] ] )</h1>
...
-            1. If _k_ > 0, then
-              1. Set _R_ to the string-concatenation of _R_ and _separator_.
-            1. Let _nextElement_ be ? Get(_array_, ! ToString(𝔽(_k_))).
-            1. If _nextElement_ is neither *undefined* nor *null*, then
-              1. Let _S_ be ? ToString(? Invoke(_nextElement_, *"toLocaleString"*)).
-              1. Set _R_ to the string-concatenation of _R_ and _S_.
+            1. If _k_ > 0, set _R_ to the string-concatenation of _R_ and _separator_.
+            1. Let _element_ be ? Get(_O_, ! ToString(𝔽(_k_))).
+            1. If _element_ is either *undefined* or *null*, let _S_ be the empty String; otherwise, let _S_ be ? ToString(? Invoke(_element_, *"toLocaleString"*)).
+            1. Set _R_ to the string-concatenation of _R_ and _S_.
```
Comment (michaelficarra) [inline spec.html:39583]: I don't like these changes. Let's change `Array.prototype.join` to match this instead.

Comment (linusg) [reply spec.html]: Sure, done :+1:

**Review by michaelficarra — APPROVED** [review]

**Comment by linusg** [comment]
Rebased on latest main - is there anything preventing this from being merged?

**Comment by michaelficarra** [comment]
@linusg Waiting on review from another editor.

**Review by bakkot — APPROVED** [review]

**Comment by bakkot** [comment]
Backed out a couple changes the editors were less convinced of.

**Comment by linusg** [comment]
Fine by me, thanks for taking a look! Rebased and cleaned up the commit history a bit :+1:

**Review by bakkot — APPROVED** [review]

**Review by syg — APPROVED** [review]

**Pull request merged**